apps/meteor/tests/e2e/omnichannel/omnichannel-tags.spec.ts

```diff
@@ -0,0 +1,159 @@
+import { createTag } from '../utils/omnichannel/tags';
+import { test, expect } from '../utils/test';
+
+test.use({ storageState: Users.admin.state });
+
+test.describe('OC - Manage Tags', () => {
+  test.skip(!IS_EE, 'OC - Manage Tags > Enterprise Edition Only');
+
+	let poOmnichannelTags: OmnichannelTags;
+
+	let department: Awaited<ReturnType<typeof createDepartment>>;
+	let agent: Awaited<ReturnType<typeof createAgent>>;
+
+	test.beforeAll(async ({ api }) => {
+		department = await createDepartment(api);
+	});
+
+	test.beforeAll(async ({ api }) => {
+		agent = await createAgent(api, 'user2');
+	});
+
+	test.afterAll(async () => {
+		await department.delete();
+		await agent.delete();
+	});
+
+  test.beforeEach(async ({ page }: { page: Page }) => {
+		poOmnichannelTags = new OmnichannelTags(page);
+	});
+
+  test('OC - Manage Tags - Create Tag', async ({ page }) => {
+    const tagName = faker.string.uuid();
+
+    await page.goto('/omnichannel');
+		await poOmnichannelTags.sidenav.linkTags.click();
+
+    await test.step('expect correct form default state', async () => {
+			await poOmnichannelTags.btnCreateTag.click();
+			await expect(poOmnichannelTags.contextualBar).toBeVisible();
+			await expect(poOmnichannelTags.btnSave).toBeDisabled();
+			await expect(poOmnichannelTags.btnCancel).toBeEnabled();
+			await poOmnichannelTags.btnCancel.click();
+			await expect(poOmnichannelTags.contextualBar).not.toBeVisible();
+		});
+
+    await test.step('expect to create new tag', async () => {
+			await poOmnichannelTags.btnCreateTag.click();
+			await poOmnichannelTags.inputName.fill(tagName);
+			await poOmnichannelTags.selectDepartment(department.data);
+			await poOmnichannelTags.btnSave.click();
+			await expect(poOmnichannelTags.contextualBar).not.toBeVisible();
+
+			await test.step('expect tag to have been created', async () => {
+				await poOmnichannelTags.search(tagName);
+				await expect(poOmnichannelTags.findRowByName(tagName)).toBeVisible();
+			});
+		});
+
+    await test.step('expect to delete tag', async () => {
+			await test.step('expect confirm delete tag', async () => {
+				await test.step('expect to be able to cancel delete', async () => {
+					await poOmnichannelTags.btnDeleteByName(tagName).click();
+					await expect(poOmnichannelTags.confirmDeleteModal).toBeVisible();
+					await poOmnichannelTags.btnCancelDeleteModal.click();
+					await expect(poOmnichannelTags.confirmDeleteModal).not.toBeVisible();
+				});
+
+				await test.step('expect to confirm delete', async () => {
+					await poOmnichannelTags.btnDeleteByName(tagName).click();
+					await expect(poOmnichannelTags.confirmDeleteModal).toBeVisible();
+					await poOmnichannelTags.btnConfirmDeleteModal.click();
+					await expect(poOmnichannelTags.confirmDeleteModal).not.toBeVisible();
+				});
+			});
+
+			await test.step('expect tag to have been deleted', async () => {
+				if (await poOmnichannelTags.inputSearch.isVisible()) {
```

Comment:
In which cases this will be false? 🤔

Reply:
Are we doing this because we are unsure about previous tags being created? Or all the tags were made within this test case? If so, I think it'd be better not to have this if, ideally we should know exactly which case would be true here.